### PR TITLE
Optimizations

### PR DESCRIPTION
**Pull request by fmkra**

### CODE REVIEW

**Review by Okm165 — APPROVED** [review]
LGTM

File: src/common/blake2s_u8.cairo [hunk 1]
```diff
@@ -52,21 +52,26 @@ fn get_sigma(r: u32) -> Array<u32> {
 }
 
 fn rotr16(n: u32) -> u32 {
-    n / 65536 + (n % 65536) * 65536
+    let (high, low) = DivRem::div_rem(n, 65536);
+    high + (low % 65536) * 65536
```
Comment (feltroidprime) [inline src/common/blake2s_u8.cairo]: The % shouldn't be needed

File: src/common/blake2s.cairo [hunk 2]
```diff
@@ -51,19 +51,23 @@ fn get_sigma(r: u32) -> Array<u32> {
 }
 
 fn rotr16(n: u32) -> u32 {
-    n / 65536 + (n % 65536) * 65536
+    let (high, low) = DivRem::div_rem(n, 65536);
+    high + (low % 65536) * 65536
```
Comment (feltroidprime) [inline src/common/blake2s.cairo]: The % shouldn't be needed

**Comment by fmkra** [comment]
I don't know what happened but code fragment you've sent isn't present on this branch. This is already fixed and looks like:
```
fn rotr12(n: u32) -> u32 {
    let (high, low) = DivRem::div_rem(n, 4096);
    TryInto::<felt252, u32>::try_into(high.into() + low.into() * 1048576).unwrap()
}
```

**Pull request merged**